### ISSUE TITLE
vscode dropped  all settings 

### ISSUE THREAD

**Issue by fragpit** [microsoft/vscode#127057]
Issue Type: <b>Bug</b>

Vscode ask me to restart, and I did. 
All settings gone.. 

VS Code version: Code 1.57.1 (507ce72a4466fbb27b715c3722558bb15afa9f48, 2021-06-17T13:28:32.912Z)
OS version: Darwin x64 20.5.0
Restricted Mode: No

maybe related #125987

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz (12 x 2600)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 5, 7|
|Memory (System)|16.00GB (0.16GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (51)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-base64|ada|0.1.0
vscode-sql-formatter|adp|1.4.4
yaml2json|ahe|0.0.3
markdown-toc|Ala|1.5.6
quitcontrol-vscode|art|3.2.1
All-Autocomplete|Ati|0.0.23
gotemplate|cas|0.4.0
gitlens|eam|11.5.1
LogFileHighlighter|emi|2.11.0
vscode-great-icons|emm|2.1.79
opensslutils|ffa|1.1.1
url-encode|fle|1.1.0
go|gol|0.26.0
vscode-systemd-support|han|0.1.1
workspacesort|ici|1.6.0
customize-ui|ioc|0.1.54
monkey-patch|ioc|0.1.15
vscode-colorize|kam|0.11.1
bash-ide-vscode|mad|1.11.0
pretty-formatter|mbl|0.2.2
prettify-json|moh|0.0.3
theme-monokai-pro-vscode|mon|1.1.19
vscode-docker|ms-|1.14.0
vscode-kubernetes-tools|ms-|1.3.3
python|ms-|2021.6.944021595
vscode-pylance|ms-|2021.6.3
jupyter|ms-|2021.6.999662501
remote-ssh|ms-|0.65.7
remote-ssh-edit|ms-|0.65.7
vscode-fluentd|msy|0.0.4
material-icon-theme|PKi|4.7.0
vscode-commons|red|0.0.6
vscode-yaml|red|0.20.0
shellman|Rem|5.5.0
partial-diff|ryu|1.4.3
jinjahtml|sam|0.16.0
vscode-sql-beautify|sen|0.0.4
vscode-nginx|sha|0.6.0
vscode-fileutils|sle|3.4.5
code-spell-checker|str|1.10.2
code-spell-checker-russian|str|0.2.10
vt100-syntax-highlighting|Tob|1.0.1
sort-lines|Tyr|1.9.0
vscode-vtools|ven|1.0.7
vscode-icons|vsc|11.5.0
codetour|vsl|0.0.56
HCL|who|0.0.5
vscode-nginx|wil|0.7.2
ansible-vault-inline|wol|0.4.0
jinja2-snippet-kit|Wya|2.0.0
vscode-ansible|zbr|0.4.5

(6 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
vspyt653:30270858
pythonvspyt602:30300191
vspor879:30202332
vspor708:30202333
vspor363:30204092
vstes627:30244334
pythonvspyt639:30300192
pythontb:30283811
pythonvspyt551cf:30311713
vspre833:30321513
pythonptprofiler:30281270
vsdfh931:30280409
vshan820:30294714
vscorecescf:30322572
pythondataviewer:30285071
vscus158:30321503
pythonvsuse255cf:30323309
vscorehov:30309549
vscod805cf:30301675
binariesv615:30325510

```

</details>

**Comment by IllusionMH** [comment]
Most likely duplicate of #125970

Did you lost your settings during update to 1.57.1 or 1.57.0 ?

**Comment by fragpit** [comment]
I'm not sure that it was an update. No release notes appeared after restart.

**Comment by sandy081** [comment]
Real Fix in #125970 covers these similar cases when extensions try to write into settings file simultaneously from multiple windows.
